### DISCUSSION TITLE
Support for Shelly dimmer 2 pushbutton dim without rule

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#13192] [Ideas]
Hi!

At this moment I'm using shelly dimmer 2 with tasmota and the following rule for pushbutton dimming:

```
Rule1
on system#boot mem1 + ENDON
ON switch1#state=2 DO power TOGGLE ENDON
ON switch1#state=4 DO dimmer %mem1% ENDON
ON switch1#state=5 DO mem1 - ENDON
ON switch1#state=6 DO mem1 + ENDON

Rule1 1
```

But when using the pushbutton, the behaviour is not really easy for non technical users.

When holding the button the brightness goes up, when releasing the button and pushing it again fast (before timeout) it dims down.

But this is difficult for some users. A more logical behavior would be to toggle the dim direction after every dim action.

Unfortunately this isn't possible without a custom build enabling logical operators in rules.

Would it be an idea to add an extra SwitchMode (16?) for dimmer pushbuttons that enables this behaviour? Also the dim dec/inc steps can be made smaller when using this switchmode giving a smoother experience.

**Comment by sfromis** [comment]
Most stuff in rules can also be be done without logical operators, but it may take tricks with extra triggers/rules/vars.

BTW. Your use of Mem variables for dimming contribute significantly to flash wear, reducing chip lifetime.

**Reply by jeroenst** [reply]
I came up with this idea:

```
Backlog SwitchMode 11; SetOption32 10;

Rule1 
on system#boot do var1 + ENDON
on switch1#state=2 do POWER TOGGLE ENDON
on switch1#state=4 do DIMMER %var1% ENDON
on switch1#state=6 do event upordown=%var1% ENDON
on event#upordown=+ do var1 - ENDON
on event#upordown=- do var1 + ENDON

Rule1 1
```

Have to test it tomorrow.

But still wouldn't be a dimmer switchmode be handy for dimmers?

**Reply by ascillato** [reply]
instead of 
```js
on system#boot var1 + ENDON`
```
it should be:
```js
Rule1 ON System#Boot DO Var1 + ENDON
```
Without the `DO`, it won't work.

**Reply by jeroenst** [reply]
thank you, I changed it.
